lib/HotModuleReplacementPlugin.js

```diff
@@ -115,11 +115,11 @@ module.exports = class HotModuleReplacementPlugin {
 					const currentChunk = compilation.chunks.find(chunk => chunk.id === chunkId);
 					if(currentChunk) {
 						const newModules = currentChunk.getModules().filter(module => module.hotUpdate);
-						const allModules = {};
-						currentChunk.forEachModule(module => {
-							allModules[module.id] = true;
-						});
-						const removedModules = records.chunkModuleIds[chunkId].filter(id => !allModules[id]);
+						const allModules = new Set();
+						for(const module of currentChunk.modulesIterable) {
+							allModules.add(module.id);
+						}
```

Comment:
```js
const allModules = new Set(Array.from(currentChunk.modulesIterable, module => module.id))
```

Is more declarative, but creates an intermediate array 🤔

The only other functional-ish way I can think of would involve writing an immediately invoked generator that has to use for..of inside it anyway so probably best to not think too deeply about it 😇